tinygrad/renderer/assembly.py

```diff
@@ -106,6 +85,18 @@ def ptr_ar(root):
       if o in u.vin and u is not n:
         u.vin = tuple(n if x == o else x for x in u.vin)
     if rew := matcher.rewrite(u): replace[u] = rew
+
+  for o,n in replace.items():
```

Comment:
What does this do? Should this be in the pattern matcher?

Reply:
This goes through the uop returned by the fxn passed into the pattern matcher and all of it's parents and adds them to the UOpGraph if they aren't there

Reply:
That should be in pattern matcher!

Reply:
oh it works on `replace` continuing on the work done in the previous block - not sure how to fit it into the pattern matcher, could you elaborate on that?